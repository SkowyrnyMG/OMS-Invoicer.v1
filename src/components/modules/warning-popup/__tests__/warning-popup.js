import React from 'react';

import {
  renderWithReduxRouter,
  screen,
  cleanup,
  waitFor,
} from 'utils/tests/test-helper';
import WarningPopup from '../warning-popup';

describe('WarningPopup', () => {
  test('should render in the document with correct styles ', async () => {
    renderWithReduxRouter(
      <WarningPopup title='test-warning-popup' isWarningOpen>
        <span>children</span>
      </WarningPopup>,
    );

    const WarningPopupNode = screen.getByTestId('warning-popup');

    await waitFor(() => {
      expect(WarningPopupNode).toHaveStyle('opacity: 1');
    });
    expect(WarningPopupNode).toMatchSnapshot();
    expect(WarningPopupNode).toBeInTheDocument();

    cleanup();
  });
});

// TODO write more edgecases!
