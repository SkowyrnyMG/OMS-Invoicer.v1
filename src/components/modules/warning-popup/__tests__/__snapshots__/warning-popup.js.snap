// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningPopup should render in the document with correct styles  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  background: #00000055;
  z-index: 50000;
  -webkit-backdrop-filter: blur(2px);
  backdrop-filter: blur(2px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  position: absolute;
  text-align: center;
  padding: 2rem 3rem;
  width: 80%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  background: #F8F8FF;
  border-radius: 5px;
}

.c2 {
  margin: 2rem;
  font-size: 24px !important;
}

.c3 {
  padding: 2rem;
  text-align: left;
}

<div
  class="c0"
  data-testid="warning-popup"
  style="opacity: 1; visibility: inherit;"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      test-warning-popup
    </h3>
    <p
      class="c3"
    >
      <span>
        children
      </span>
    </p>
  </div>
</div>
`;
